### PR TITLE
Rename commands to buildtest buildspec edit-test and buildtest buildspec edit-file

### LINKED ISSUE #1080: [FEATURE]: Rename buildtest buildspec edit
### Please describe your feature request

Let's rename `buildtest buildspec edit` to `buildtest buildspec edit-test`

- [x] Update option in https://github.com/buildtesters/buildtest/blob/devel/buildtest/cli/__init__.py
- [x] Update bash completion https://github.com/buildtesters/buildtest/blob/devel/bash_completion.sh
- [x] Update the user documentation https://github.com/buildtesters/buildtest/blob/devel/docs/gettingstarted/buildspecs_interface.rst#editing-buildspecs-in-your-preferred-editor
- [ ] Add regression test for https://app.codecov.io/gh/buildtesters/buildtest/blob/devel/buildtest/cli/buildspec.py or ignore the method during coverage result using `exclude_lines` see https://coverage.readthedocs.io/en/latest/excluding.html 

### Post question in Slack

- [X] I agree that I posted my question in [slack](https://hpcbuildtest.slack.com/) before creating this issue

### Is there an existing issue

- [X] I confirm there is no existing [issue](https://github.com/buildtesters/buildtest/issues) for this issue

### PR DESCRIPTION
**Pull request by JeffreyFG**
Changed the documentation to reflect the change to the ``buildtest buildspec edit-test`` command. The alias for the buildspec command that allowed you to edit tests based on a supplied test name has been changed from ``buildtest buildspec edit`` to ``buildtest buildspec edit-test``.  This was done in order to better match with the ``buildtest buildspec edit-file``.

### CODE REVIEW

File: docs/gettingstarted/buildspecs_interface.rst [hunk 1]
```diff
@@ -421,12 +421,12 @@ Edit buildspecs ``buildtest buildspec edit-file``
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The ``buildtest buildspec edit-file`` command can be used to edit buildspec based on filename as pose to testname.
-This command works similar to ``buildtest buildspec edit`` where each file is open in editor and validated upon completion.
-You can use this command to create new buildspec whereas ``buildtest buildspec edit`` only works on existing buildspecs loaded
+This command works similar to ``buildtest buildspec edit-test`` where each file is open in editor and validated upon completion.
```
Comment (shahzebsiddiqui) [inline docs/gettingstarted/buildspecs_interface.rst:424]: I think you mean `buildtest buildspec edit-file` here since this is section on editing by buildspecs. The section above is for `edit-test`

File: docs/gettingstarted/buildspecs_interface.rst [hunk 2]
```diff
@@ -421,12 +421,12 @@ Edit buildspecs ``buildtest buildspec edit-file``
...
+You can use this command to create new buildspec whereas ``buildtest buildspec edit-test`` only works on existing buildspecs loaded
```
Comment (shahzebsiddiqui) [inline docs/gettingstarted/buildspecs_interface.rst:425]: Same here please change to `edit-file`

**Comment by shahzebsiddiqui** [comment]
@JeffreyFG thanks for your contribution. I presume you are going to make the change to codebase first before updating the docs. Can you please make the changes in this same PR.

**Comment by JeffreyFG** [comment]
> @JeffreyFG thanks for your contribution. I presume you are going to make the change to codebase first before updating the docs. Can you please make the changes in this same PR.

I have made changes to the code base but i had questions on the proper etiquette for pushing those change so i haven't done so yet

File: bash_completion.sh [hunk 3]
```diff
@@ -83,7 +83,7 @@ _buildtest ()
 
   COMPREPLY=()   # Array variable storing the possible completions.
 
-  local cmds="build buildspec cd cdash clean config debugreport docs edit help inspect history path report schema schemadocs stylecheck unittests"
+  local cmds="build buildspec cd cdash clean config debugreport docs edit-test help inspect history path report schema schemadocs stylecheck unittests"
```
Comment (shahzebsiddiqui) [inline bash_completion.sh]: please remove `edit-test` from here these are the top level commands  such as `buildtest build`. The `buildtest buildspec edit-test` is a subcommand which is followed in separate case statement.

File: bash_completion.sh [hunk 4]
```diff
@@ -158,7 +158,7 @@ _buildtest ()
       esac
       ;;
     config|cg)
-      local cmds="-h --help compilers edit executors validate view systems"
+      local cmds="-h --help compilers edit-test executors validate view systems"
```
Comment (shahzebsiddiqui) [inline bash_completion.sh]: This is the section for `buildtest config` and we have command `buildtest config edit` which is a valid command. Please revert change

File: bash_completion.sh [hunk 5]
```diff
@@ -216,7 +216,7 @@ _buildtest ()
       ;;
 
     buildspec|bc)
-      local cmds="-h --help edit edit-file find show summary validate"
+      local cmds="-h --help edit-test edit-file find show summary validate"
```
Comment (shahzebsiddiqui) [inline bash_completion.sh:219]: This is correct place to make change! Good job. 

You can test bash completion by doing `source bash_completion.sh` and then `TAB` away :-)

File: buildtest/cli/__init__.py [hunk 6]
```diff
@@ -497,7 +497,7 @@ def buildspec_menu(subparsers):
     )
 
     edit_buildspecs = subparsers_buildspec.add_parser(
-        "edit", help="Edit buildspec file based on test name"
+        "edit-test", help="Edit buildspec file based on test name"
```
Comment (shahzebsiddiqui) [inline buildtest/cli/__init__.py:500]: We want this to be `edit-file` not `edit-test`

File: buildtest/cli/__init__.py [hunk 7]
```diff
@@ -658,7 +658,7 @@ def config_menu(subparsers):
 
     compilers = subparsers_config.add_parser("compilers", help="Search compilers")
 
-    subparsers_config.add_parser("edit", help="Open configuration file in editor")
+    subparsers_config.add_parser("edit-test", help="Open configuration file in editor")
```
Comment (shahzebsiddiqui) [inline buildtest/cli/__init__.py]: So this method `config_menu` is for building the arguments for `buildtest config` and we want to keep `buildtest config edit` command so please revert this change.

**Comment by shahzebsiddiqui** [comment]
@JeffreyFG this looks great! Thanks for your contribution to buildtest.
